nemo/collections/nlp/models/language_modeling/lm_model.py

```diff
@@ -51,34 +53,42 @@ def output_types(self) -> Optional[Dict[str, NeuralType]]:
             'nsp_logits': self.nsp_classifier.output_types['logits'],
         }
 
-    def __init__(
-        self,
-        config_file: Optional[str] = None,
-        pretrained_model_name: Optional[str] = 'bert-base-uncased',
-        preprocessing_args: Optional[Dict[str, str]] = None,
-    ):
-        """
-        Args:
-            config_file: model config file
-            pretrained_model_name: BERT model name 
-            preprocessing_args: preprocessing parameters for on the fly data preprocessing
-        """
-        # TODO: This is a workaround - please fix - see text_classification_model or asr for example
-        cfg = DictConfig(
-            {
-                'config_file': config_file,
-                'pretrained_model_name': pretrained_model_name,
-                'preprocessing_args': preprocessing_args,
-            }
-        )
-        super().__init__(cfg=cfg)
-        self.pretrained_model_name = pretrained_model_name
-        self.tokenizer = None
-        if preprocessing_args:
-            self.__setup_tokenizer(preprocessing_args)
-        self.bert_model = get_pretrained_lm_model(pretrained_model_name=pretrained_model_name, config_file=config_file)
+    def __init__(self, cfg: DictConfig, trainer: Trainer = None):
+        if cfg.language_model.bert_config_file is not None:
+            logging.info(
```

Comment:
How do you know file exists?
If we change the order of tokenizer and get_model, then there is no need for this block.

Reply:
cfg.language_model.bert_config_file exists because it in the .yaml and we have strict=true by default with hydra.

Reply:
Switching the order is fine, but then it means we have to call super().__init__(cfg=cfg, trainer=trainer) later, because our datasets need the tokenizer.

Reply:
So I've switched the order, but I think this is a good thing to let the user know they are loading from a file.

Reply:
Okay, so when I switched the order, I get this error, which I guess is why I had it in this order in the first place:
```
Traceback (most recent call last):
  File "bert_pretraining_from_text.py", line 30, in main
    bert_model = BERTMLMModel(cfg.model, trainer=trainer)
  File "/home/eharper/github/new_NeMo/NeMo/nemo/collections/nlp/models/language_modeling/lm_model.py", line 73, in __init__
    config_dict=OmegaConf.to_container(cfg.language_model.bert_config),
  File "/opt/conda/lib/python3.6/site-packages/torch/nn/modules/module.py", line 650, in __setattr__
    "cannot assign module before Module.__init__() call")
AttributeError: cannot assign module before Module.__init__() call
```
So I think the way it was before was better, since it was working.